lib/demo/accounts/user.ex

```diff
@@ -0,0 +1,119 @@
+defmodule Demo.Accounts.User do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "users" do
+    field :email, :string
+    field :password, :string, virtual: true
+    field :hashed_password, :string
+    field :confirmed_at, :naive_datetime
+
+    timestamps()
+  end
+
+  @doc """
+  A user changeset for registration.
+
+  It is important to validate the length of both e-mail and password.
+  Otherwise databases may truncate them without warnings, which could
+  lead to unpredictable or insecure behaviour. Long passwords may also
+  be very expensive to hash.
+  """
```

Comment:
I'm thinking a lot of the functions in this module are internal APIs and these functions shouldn't be showing up in ExDoc... What do you think about switching these `@doc` strings to comments so the user can differentiate public API functions from internal functions like these?

Reply:
I think in most cases the same team is maintaining and using both private and public APIs, so it doesn’t really matter. I feel like using @doc mostly for consistency across all generated files. @chrismccord, do you have an opinion on this?